src/vulkan-renderer/world/collision_query.cpp

```diff
@@ -0,0 +1,107 @@
+#include "inexor/vulkan-renderer/world/collision_query.hpp"
+
+#include <glm/gtx/intersect.hpp>
+
+namespace inexor::vulkan_renderer::world {
+
+OctreeCollisionQuery::OctreeCollisionQuery(const Cube &cube) : m_cube(cube) {}
+
+bool OctreeCollisionQuery::ray_box_collision(const std::array<glm::vec3, 2> box_bounds, const glm::vec3 position,
+                                             const glm::vec3 direction) const {
+    glm::vec3 inverse_dir{1 / direction.x, 1 / direction.y, 1 / direction.z};
+    std::int32_t sign[3]{inverse_dir.x < 0, inverse_dir.y < 0, inverse_dir.z < 0};
+
+    float tmin, tmax, tymin, tymax, tzmin, tzmax;
+
+    tmin = (box_bounds[sign[0]].x - position.x) * inverse_dir.x;
+    tmax = (box_bounds[1 - sign[0]].x - position.x) * inverse_dir.x;
+    tymin = (box_bounds[sign[1]].y - position.y) * inverse_dir.y;
+    tymax = (box_bounds[1 - sign[1]].y - position.y) * inverse_dir.y;
+
+    if ((tmin > tymax) || (tymin > tmax)) {
+        return false;
+    }
+    if (tymin > tmin) {
+        tmin = tymin;
+    }
+    if (tymax < tmax) {
+        tmax = tymax;
+    }
+
+    tzmin = (box_bounds[sign[2]].z - position.z) * inverse_dir.z;
+    tzmax = (box_bounds[1 - sign[2]].z - position.z) * inverse_dir.z;
+
+    if ((tmin > tzmax) || (tzmin > tmax)) {
```

Comment:
Lines 34 -38 can be simplified: return !((tmin > tzmax) || (tzmin > tmax));

Reply:
Done.